src/scratch_comment_bubble.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from "blockly/core";
+
+/**
+ * A Scratch-style comment bubble for block comments.
+ * @implements {IBubble}
+ * @implements {ISelectable}
+ */
+export class ScratchCommentBubble extends Blockly.comments.CommentView {
+  constructor(sourceBlock) {
+    super(sourceBlock.workspace);
+    this.sourceBlock = sourceBlock;
+    this.disposing = false;
+    this.id = Blockly.utils.idGenerator.genUid();
+    this.getSvgRoot().setAttribute(
+      "style",
+      `--colour-commentBorder: ${sourceBlock.getColourTertiary()};`
+    );
+
+    Blockly.browserEvents.conditionalBind(
+      this.getSvgRoot(),
+      "pointerdown",
+      this,
+      this.startGesture
+    );
+    // Don't zoom with mousewheel; let it scroll instead.
+    Blockly.browserEvents.conditionalBind(
+      this.getSvgRoot(),
+      "wheel",
+      this,
+      (e) => {
+        e.stopPropagation();
+      }
+    );
+  }
+
+  setDeleteStyle(enable) {}
+  showContextMenu() {}
+  setDragging(start) {}
+  select() {}
+  unselect() {}
+
+  isMovable() {
+    return true;
+  }
+
+  moveDuringDrag(newLocation) {
+    this.moveTo(newLocation);
+  }
+
+  moveTo(xOrCoordinate, y) {
```

Comment:
Should we be firing move events from here? or only from the `endDrag` method?

[Edit: ah wait, the events are also being fire from `setBubbleLocation` so maybe fine?]

Reply:
Yeah as you noted this should be fine, and firing them (and making the VM update its state) continuously during a drag seems excessive.